src/main/java/com/iota/iri/service/snapshot/impl/SnapshotStateImpl.java

```diff
@@ -166,7 +166,9 @@ public boolean equals(Object obj) {
         HashMap<Hash, Long> result = new HashMap<>();
         balances.forEach((key, value) -> {
             if (value < 0) {
-                log.info("negative value for address " + key + ": " + value);
+                if (log.isDebugEnabled()) {
```

Comment:
Whats the reason for enclosing it in the debugEnabled? This is generally used to prevent expensive method calls from running when the output of that call is only used in logging.

But since we dont do that, and log to debug now, it seems kind of double

Reply:
`"negative value for address " + key + ": " + value`

Here we are concatenating strings and creating new objects which won't be used if we are not in debug mode

Reply:
Id rather see this then:
`log.debug("negative value for address {}: {}", key, value);`

Reply:
Yup this is the correct way to do it

Reply:
@GalRogozinski You still have the isDebugEnabled check now :)

Reply:
fixed